### ISSUE TITLE
Plans to move to V3 of ModSecurity?

### ISSUE THREAD

**Issue by sealneaward** [jcmoraisjr/modsecurity-spoa#10]
We are observing serious issues with memory consumption on V2 of ModSecurity in our kubernetes clusters.
We run performance tests against our application deployments, the ModSecurity containers just acrue memory to absurd levels and no garbage collection is done once the performance test is finished.

From what I've read on different issues pages, it looks like there are a lot of memory leaks that can occur in V2 of ModSecurity that are fixed in V3.
I was wondering if there are plans to upgrade the ModSecurity version to V3?

**Comment by jcmoraisjr** [comment]
Yep, this should be as simple as update dependencies but the problem is the existence of a good wrapper. HAProxy's one is good enough but V2 only. A co worker [has started](https://github.com/rikatz/ingress-security-agent/) an attempt to write one but didn't evolve. This might be my next challenge in the near future, and let me know if you find a good wrapper out there that talks HAProxy's spop.

**Comment by sealneaward** [comment]
Hmm. I don't think I have the expertise to write a wrapper for V3. 
I'm testing a more kubernetes native approach to the memory leaks in V2 of ModSecurity.
I think once I confirm it works, I can bring it into a docs change for HAProxy.
Largely it just setting resource limits to kill the ModSecurity process on the container and adding a liveness probe for the default port to restart the pod if ModSecurity is not available.

**Comment by CorentinS6** [comment]
Hello,
Have any of you tried this? > https://github.com/corazawaf/coraza-spoa
Regards

**Comment by sealneaward** [comment]
Yes, I have actually tested it for the ingress controller that @jcmoraisjr maintains. It works.
We even added [documentation](https://haproxy-ingress.github.io/docs/examples/modsecurity/#using-coraza-instead-of-modsecurity) on how to swap it in place.
